util/test/account_testutil.go

```diff
@@ -99,19 +101,19 @@ func MakeAssetConfigOrPanic(round, configid, assetid, total, decimals uint64, de
 }
 
 // MakeAssetFreezeOrPanic create an asset freeze/unfreeze transaction.
-func MakeAssetFreezeOrPanic(round, assetid uint64, frozen bool, addr types.Address) (*types.SignedTxnWithAD, *idb.TxnRow) {
+func MakeAssetFreezeOrPanic(round, assetid uint64, frozen bool, send, fadd types.Address) (*types.SignedTxnWithAD, *idb.TxnRow) {
```

Comment:
In this code, there is no reason we can't use the full name. For the encoded values like `snd` and `fadd` it is to save space.

Why protobuf instead of cap'n proto? There are lots of them, I'm not sure why msgp was chosen. It looks like it is one of the best regarding space efficiency according to this project: https://github.com/thekvs/cpp-serializers

Reply:
I'd vote for protobuf, but it has one key deficiency: it refuses to guarantee that it will always serialize fields in the same order. That makes it unusable for anything that needs a strict canonical serialization that can be cryptographically signed or verified. Though, we hacked our msgpack implementation to ensure sorted keys and a couple other things to gain that requirement. We could have forked anything else and gotten the same but better encoding. I did some early testing that says blocks could be 20% smaller protobuf encoded vs msgpack; but dropped it because we have too much weight of existing code and infrastructure to change lightly. Some time in the next year as we try to push to 10x and 47x performance we may have to reevaluate and use a more efficient encoding.